pkg/proxy/provider/http/http.go

```diff
@@ -0,0 +1,164 @@
+type Provider struct {
+	Endpoint     string         `description:"Load configuration from this endpoint." json:"endpoint" toml:"endpoint" yaml:"endpoint" export:"true"`
+	PollInterval types.Duration `description:"Polling interval for endpoint." json:"pollInterval,omitempty" toml:"pollInterval,omitempty" yaml:"pollInterval,omitempty"`
+	PollTimeout  types.Duration `description:"Polling timeout for endpoint." json:"pollTimeout,omitempty" toml:"pollTimeout,omitempty" yaml:"pollTimeout,omitempty"`
+}
+
+// New creates a new instance of the HTTP provider.
+func New(endpoint string, pollInterval types.Duration, pollTimeout types.Duration) *Provider {
+	return &Provider{
+		Endpoint:     endpoint,
+		PollInterval: pollInterval,
+		PollTimeout:  pollTimeout,
+	}
+}
+
+// Init the provider.
+func (p *Provider) Init() error {
+	if len(p.Endpoint) == 0 {
+		return fmt.Errorf("a non-empty endpoint is required")
+	}
+
+	if p.PollInterval == 0 {
+		p.PollInterval = types.Duration(15 * time.Second)
+	}
+
+	if p.PollTimeout == 0 {
+		p.PollTimeout = types.Duration(15 * time.Second)
+	}
+
+	return nil
+}
+
+// Provide allows the provider to provide configurations to traefik
+// using the given configuration channel.
+func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.Pool) error {
+	pool.GoCtx(func(routineCtx context.Context) {
+		ctxLog := log.With(routineCtx, log.Str(log.ProviderName, "http"))
+		logger := log.FromContext(ctxLog)
+
+		operation := func() error {
+			errChan := make(chan error)
+			ticker := time.NewTicker(time.Duration(p.PollInterval))
+
+			pool.GoCtx(func(ctx context.Context) {
+				ctx = log.With(ctx, log.Str(log.ProviderName, "http"))
+
+				defer close(errChan)
+				for {
+					select {
+					case <-ticker.C:
+						data, err := p.getDataFromEndpoint(ctxLog)
+						if err != nil {
+							logger.Errorf("Failed to get config from endpoint: %w", err)
+							errChan <- err
+							return
+						}
+
+						configuration := p.buildConfiguration(ctx, data)
+						if configuration != nil {
+							configurationChan <- dynamic.Message{
+								ProviderName:  "http",
+								Configuration: configuration,
+							}
+						}
+
+					case <-ctx.Done():
+						ticker.Stop()
+						return
+					}
+				}
+			})
+			if err, ok := <-errChan; ok {
+				return err
+			}
+			// channel closed
+			return nil
+		}
+
+		notify := func(err error, time time.Duration) {
+			logger.Errorf("Provider connection error %w, retrying in %s", err, time)
+		}
+		err := backoff.RetryNotify(safe.OperationWithRecover(operation), backoff.WithContext(job.NewBackOff(backoff.NewExponentialBackOff()), ctxLog), notify)
+		if err != nil {
+			logger.Errorf("Cannot connect to http server %w", err)
+		}
+	})
+
+	return nil
+}
+
+// getDataFromEndpoint gets data from the configured provider endpoint, and returns the data.
+func (p *Provider) getDataFromEndpoint(ctx context.Context) ([]byte, error) {
```

Comment:
```suggestion
func (p Provider) getDataFromEndpoint(ctx context.Context) ([]byte, error) {
```

The pointer receiver does not seem necessary